students/kmsnyde/lesson09/mailroom5_classes.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun May 20 07:03:34 2018
+
+@author: Karl M. Snyder
+"""
+
+class Main:
+    def __init__(self):
+        self.menu = {1: 'Send a Thank You',
+                     2: 'Create a Report',
+                     3: 'Send letters to everyone',
+                     4: 'Quit'}
+    
+    def main_menu(self):
+        print('\n', 'Please select a number from the following choices:\n')
+        return {(print(str(k) + ':', v)) for k, v in self.menu.items()}
+    
+    def selection(self):
+        while True:
+            input1 = input("Selection: ")
+            #print(type(input1), input1)
+            try:
+                if int(input1) in range(1, 4):
+                    if int(input1) == 1:
+                        print('\nType a user\'s name or "list" to show names.')
+                        input2 = input('-> ')
+                        if input2 == 'list':
+                            our_donors.donors()
+                            continue
+                        else:
+                            input3 = input('Donation amount: ')
+                            user = Donor(input2, input3)
+                            user.send_thanks()
+                    if int(input1) == 2:
+                        our_donors.create_report()
+                    if int(input1) == 3:
+                        our_donors.send_letters_all()
+                elif int(input1) == 4:
+                    print("Exiting program...")
+                    break
+            except ValueError:
+                print("You must use a menu number between 1-4; try again!")
+
+class Donor:
```

Comment:
your `Donor` class should represent a single `Donor` with properties like `name` (string) and `donations` (list) at the least.

your `Donors` should handle a collection of `Donor` objects, internally it may create a dictionary to make it easier.

here is an example of interaction:
```
donor1 = Donor("Cat", [100])
donor2 = Donor("Bob", [200])
donors = Donors(donor1, donor2)
donors.create_report()
```
----
your current `Donor` has knowledge of many donors which shouldn't be the case.